addons/mail/controllers/discuss/channel.py

```diff
@@ -141,3 +141,27 @@ def discuss_channel_fold(self, channel_id, state, state_count):
         if not member:
             raise NotFound()
         return member._channel_fold(state, state_count)
+
+    @http.route("/discuss/channel/sub_channel/fetch", methods=["POST"], type="json", auth="public")
+    @add_guest_to_context
+    def discuss_channel_sub_channel_fetch(self, parent_channel_id, search_term=None, before=None, limit=30):
+        channel = request.env["discuss.channel"].search([("id", "=", parent_channel_id)])
+        if not channel._find_or_create_member_for_self():
+            raise NotFound()
+        domain = [("parent_channel_id", "=", channel.id)]
+        if before:
+            domain.append(("id", "<", before))
+        if search_term:
+            domain.append(("name", "ilike", search_term))
+        sub_channels = request.env["discuss.channel"].search(domain, order="id desc", limit=limit)
+        store = Store(sub_channels)
+        store.add(sub_channels._get_last_messages())
+        return store.get_result()
+
+    @http.route("/discuss/channel/sub_channel/join", methods=["POST"], type="json", auth="public")
+    @add_guest_to_context
+    def discuss_channel_sub_channel_join(self, sub_channel_id):
+        if sub_channel := request.env["discuss.channel"].browse(sub_channel_id).exists():
```

Comment:
Why not just `search`?

Reply:
In low level code working with recordset and ids yes to avoid queries, but this clearly wants to make a query to find the channel, `search` is made for that.

On top of that, that browse.exists does not check ACL, it only fails if the following code tries to read an inaccessible field, but then it will crash rather than appropriately being caught as "not channel".

And if it's only followed by an existence check and never reading anything else the code will (probably incorrectly) accept the channel.